src/Plugin/GraphQL/DataProducer/Entity/EntityQuery.php

```diff
@@ -156,46 +99,22 @@ public function __construct(
    * @throws \GraphQL\Error\UserError
    *   No bundles defined for given entity type.
    */
-  public function resolve(string $type, int $limit = 10, ?int $offset, ?bool $ownedOnly, ?array $conditions, ?array $allowedFilters, ?string $language, ?array $bundles, ?array $sorts, FieldContext $context): array {
-    // Make sure offset is zero or positive.
-    $offset = max($offset ?: 0, 0);
-
-    $entity_type = $this->entityTypeManager->getStorage($type);
-    $query = $entity_type->getQuery()
-      ->range($offset, $limit);
-
-    // Query only those entities which are owned by current user, if desired.
-    if ($ownedOnly) {
-      $query->condition('uid', $this->currentUser->id());
-      // Add user cacheable dependencies.
-      $account = $this->currentUser->getAccount();
-      $context->addCacheableDependency($account);
-      // Cache response per user to make sure the user related result is shown.
-      $context->addCacheContexts(['user']);
-    }
-
-    // Ensure that access checking is performed on the query.
-    $query->accessCheck(TRUE);
-
-    if (isset($bundles)) {
-      $bundle_key = $entity_type->getEntityType()->getKey('bundle');
-      if (!$bundle_key) {
-        throw new UserError('No bundles defined for given entity type.');
-      }
-      $query->condition($bundle_key, $bundles, "IN");
-    }
-    if (isset($language)) {
-      $query->condition('langcode', $language);
-    }
+  public function resolve(string $type, int $limit, int $offset, bool $ownedOnly, array $conditions, array $allowedFilters, array $languages, array $bundles, array $sorts, FieldContext $context): array {
+    $query = $this->buildBaseEntityQuery(
+      $type,
+      $ownedOnly,
+      $conditions,
+      $allowedFilters,
+      $languages,
+      $bundles,
+      $context
+    );
 
-    foreach ($conditions as $condition) {
-      if (!in_array($condition['field'], $allowedFilters)) {
-        throw new UserError("Field '{$condition['field']}' is not allowed as filter.");
-      }
-      $operation = isset($condition['operator']) ? $condition['operator'] : NULL;
-      $query->condition($condition['field'], $condition['value'], $operation);
-    }
+    // Make sure offset is zero or positive.
+    $offset = max($offset, 0);
+    $query->range($offset, $limit);
```

Comment:
we should have an upper max limit to prevent DoS attacks. What do we usually use, 100?